content/docs/cucumber/configuration.md

```diff
@@ -15,170 +15,376 @@ polyglot:
+
+        ParameterType("book", ".*") { s : String ->
+            Book(s)
         }
 
-        @Override
-        public <T> T transform(String s, Class<T> aClass) {
-            return objectMapper.convertValue(s, aClass);
+        DataTableType { entry: Map<String, String> ->
+            Author(entry["firstName"], entry["lastName"], entry["famousBook"])
         }
+
+        DocStringType("json") { docString: String -> 
+	        objectMapper.readValue(docString, JsonNode::class)
+	    }
     }
 }
 ```
+{{% /block %}}
+
+{{% block "java,kotlin" %}}
+Using the `@DefaultParameterTransformer`, `@DefaultDataTableEntryTransformer`
+and `DefaultDataTableCellTransformer` annotations also possible to plugin an 
+ObjectMapper. The object mapper (Jackson in this example) will handle the 
+conversion of anonymous parameter types and data table entries.
+{{% /block %}}
+
+{{% block "java" %}}
+
+```java
+package com.example;
 
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.cucumber.java.DefaultDataTableCellTransformer;
+import io.cucumber.java.DefaultDataTableEntryTransformer;
+import io.cucumber.java.DefaultParameterTransformer;
+
+import java.lang.reflect.Type;
+
+public class StepDefinitions {
+
+    private final ObjectMapper objectMapper = new ObjectMapper();
+
+    @DefaultParameterTransformer
+    @DefaultDataTableEntryTransformer
+    @DefaultDataTableCellTransformer
+    public Object transformer(Object fromValue, Type toValueType) {
+        return objectMapper.convertValue(fromValue, objectMapper.constructType(toValueType));
+    }
+}
+```
+{{% /block %}}
 
+{{% block "kotlin" %}}
 ```kotlin
 package com.example
 
-import io.cucumber.core.api.TypeRegistryConfigurer
-import io.cucumber.core.api.TypeRegistry
-import io.cucumber.cucumberexpressions.ParameterByTypeTransformer
-import io.cucumber.datatable.TableCellByTypeTransformer
-import io.cucumber.datatable.TableEntryByTypeTransformer
 import com.fasterxml.jackson.databind.ObjectMapper
+import io.cucumber.java.DefaultDataTableCellTransformer
+import io.cucumber.java.DefaultDataTableEntryTransformer
+import io.cucumber.java.DefaultParameterTransformer
 
 import java.lang.reflect.Type
-import java.util.Locale
-import java.util.Map
 
-import static java.util.Locale.ENGLISH
+class StepDefinitions {
 
-class TypeRegistryConfiguration : TypeRegistryConfigurer {
+    private val objectMapper = ObjectMapper()
 
-    override fun locale(): Locale {
-        return ENGLISH
+    @DefaultParameterTransformer
+    @DefaultDataTableEntryTransformer
+    @DefaultDataTableCellTransformer
+    fun transformer(fromValue: Any, toValueType: Type): Any {
+        return objectMapper.convertValue(fromValue, objectMapper.constructType(toValueType))
     }
+}
+```
+{{% /block %}}
+
+{{% block "java,kotlin" %}}
+For lambda defined step definitions, there are `DefaultParameterTransformer`, 
+`DefaultDataTableCellTransformer` and `DefaultDataTableEntryTransformer` functions:
```

Comment:
```suggestion
`DefaultDataTableCellTransformer` and `DefaultDataTableEntryTransformer` {{% stepdef-body %}}:
```

Reply:
Use this short code to correctly display "method" or "function" for the respective programming language. (Although I'm not sure if/how that would work with a text for 2 different languages)